### PR TITLE
[Android] - Add file type support and update core to 7.0.2 SDK-4096

### PR DESCRIPTION
**Pull request by piyush-kukadiya**
- [x] Add `CleverTap.defineFileVariable("fileVariable")`
- [x] Add `CleverTap.onOneTimeVariablesChanged(handler)`
- [x] Add `CleverTap.onVariablesChangedAndNoDownloadsPending(handler)`
- [x] Add `CleverTap.onceVariablesChangedAndNoDownloadsPending(handler)`
- [x] Add `CleverTap.onFileValueChanged('fileVariable', handler)`
- [x] Support core SDK `7.0.2`
- [x] Update [Variables.md](https://github.com/CleverTap/clevertap-react-native/blob/master/docs/Variables.md)
- [ ] Update [ChangeLog.md](https://github.com/CleverTap/clevertap-react-native/blob/master/CHANGELOG.md)
- [x] Update Example App
- [x] Bump lib version to `3.1.0`

### CODE REVIEW

**Review by kushCT — APPROVED** [review]

**Comment by kushCT** [comment]
@piyush-kukadiya I have pushed the changes from iOS which can be directly merged into develop branch from here. We don't need to merge the changes back to it's parent branch.

File: src/index.js [hunk 1]
```diff
@@ -918,8 +931,8 @@ var CleverTap = {
     * Create File Variable
     * @param {string} fileVariable - the file variable string
     */
-        defineFileVariables: function (fileVariable) {
-            CleverTapReact.defineFileVariables(fileVariable)
+        defineFileVariable: function (fileVariable) {
```
Comment (Anush-Shand) [inline src/index.js:934]: Formatting is a little off here

**Review by nzagorchev — CHANGES_REQUESTED** [review]
Looks good, dev tested it quickly on iOS. A few tiny changes requested.

File: src/index.js (hunk 1 above)
Comment (kushCT) [reply src/index.js]: Resolved

File: ios/CleverTapReact/CleverTapReact.mm [hunk 2]
```diff
@@ -1065,22 +1074,29 @@ - (CTLocalInApp*)_localInAppConfigFromReadableMap: (NSDictionary *)json {
...
+    RCTLogInfo(@"[CleverTap onVariablesChangedAndNoDownloadsPending]");
     [[self cleverTapInstance]onVariablesChangedAndNoDownloadsPending:^{
-        [[NSNotificationCenter defaultCenter] postNotificationName:kCleverTapOnFileVariablesChangedAndNoDownloadsPending object:nil userInfo:[self getVariableValues]];
+        [[NSNotificationCenter defaultCenter] postNotificationName:kCleverTapOnVariablesChangedAndNoDownloadsPending object:nil userInfo:[self getVariableValues]];
+    }];
+}
+
+RCT_EXPORT_METHOD(onceVariablesChangedAndNoDownloadsPending) {
+    RCTLogInfo(@"[CleverTap onceVariablesChangedAndNoDownloadsPending]");
+    [[self cleverTapInstance]onceVariablesChangedAndNoDownloadsPending:^{
```
[the remark this answers is not in the document]
Comment (kushCT) [reply ios/CleverTapReact/CleverTapReact.mm]: Resolved

File: ios/CleverTapReact/CleverTapReact.mm [hunk 3]
```diff
@@ -1036,12 +1038,12 @@ - (CTLocalInApp*)_localInAppConfigFromReadableMap: (NSDictionary *)json {
...
 }
 
-RCT_EXPORT_METHOD(defineFileVariables:(NSString*)fileVariables) {
-    RCTLogInfo(@"[CleverTap defineFileVariables]");
-    if (!fileVariables) return;
-    CTVar *fileVar = [[self cleverTapInstance]defineFileVar:fileVariables];
+RCT_EXPORT_METHOD(defineFileVariable:(NSString*)fileVariable) {
+    RCTLogInfo(@"[CleverTap defineFileVariable]");
+    if (!fileVariable) return;
+    CTVar *fileVar = [[self cleverTapInstance]defineFileVar:fileVariable];
```
[the remark this answers is not in the document]
Comment (kushCT) [reply ios/CleverTapReact/CleverTapReact.mm]: Resolved

File: Example/app/App.js [hunk 4]
```diff
@@ -791,19 +799,32 @@ export default class App extends Component {
...
+        break;
+      case Actions.FILES_VARIABLES_CHANGED_AND_DOWNLOADED_ONCE:
+        CleverTap.onceVariablesChangedAndNoDownloadsPending((variables) => {
+          console.log('onceVariablesChangedAndNoDownloadsPending', variables);
         });
         break;
       case Actions.FILE_CHANGED:
-      CleverTap.onFileChanged('fileVariable', (variable) => {
-        console.log('onFileChanged: ', variable);
+      CleverTap.onFileValueChanged('pk.fileVariable', (variable) => {
```
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/app/App.js]: Resolved

File: Example/app/App.js [hunk 5]
```diff
@@ -791,19 +799,32 @@ export default class App extends Component {
...
+      case Actions.ONE_TIME_VARIABLES_CHANGED:
+        CleverTap.onOneTimeVariablesChanged((variables) => {
+          console.log('onOneTimeVariablesChanged: ', variables);
+        });
+        break;
       case Actions.VALUE_CHANGED:
         CleverTap.onValueChanged('reactnative_var_string', (variable) => {
           console.log('onValueChanged: ', variable);
         });
+        CleverTap.onValueChanged('pk.fileVariable', (variable) => {
```
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/app/App.js]: Resolved

**Pull request merged**